Client/Frontend/Browser/BrowserViewController/BVC+Rewards.swift

```diff
@@ -115,18 +115,41 @@ extension Tab {
     func reportPageLoad(to rewards: BraveRewards) {
         guard let webView = webView, let url = webView.url else { return }
         if url.isLocal || PrivateBrowsingManager.shared.isPrivateBrowsing { return }
+                
+        var htmlBlob: String?
+        var classifierText: String?
         
-        let getHtmlToStringJSCall = "document.documentElement.outerHTML.toString()"
-        // Copy to var, as `shouldClassifyLoadsForAds` can be reset before JS completes
-        let shouldClassify = shouldClassifyLoadsForAds
-        webView.evaluateJavaScript(getHtmlToStringJSCall, completionHandler: { html, _ in
+        let group = DispatchGroup()
+        group.enter()
+        webView.evaluateJavaScript("document.documentElement.outerHTML.toString()", completionHandler: { html, _ in
+            defer { group.leave() }
             guard let htmlString = html as? String else { return }
+            htmlBlob = htmlString
+        })
+        
+        if shouldClassifyLoadsForAds {
+            group.enter()
+            webView.evaluateJavaScript("document.body.innerText", completionHandler: { text, _ in
+                defer { group.leave() }
+                guard var innerText = text as? String else { return }
+                for c in ["\'", "\"", "\\", "\n", "\r", "\t", "\0"] {
```

Comment:
Could add some documentation, I am not sure what are these all characters for :D and whether this is a full list

Reply:
This is the list Terry gave me, I'll add a comment but these are a bunch of the escape characters that they don't want in the classifier 🤷‍♂️

Reply:
that's enough for a comment in code for me :)